catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SeedCommand.java

```diff
@@ -0,0 +1,195 @@
+        }
+    }
+
+    @Override
+    public Object execute() throws Exception {
+        return doExecute();
+    }
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        if (productLimit <= 0) {
+            printErrorMessage("A limit of " + productLimit
+                    + " was supplied. The limit must be greater than 0.");
+            return null;
+        }
+
+        final Filter filter = createFilter();
+
+        final long start = System.currentTimeMillis();
+        int productDownloads = 0;
+        int downloadErrors = 0;
+        int pageCount = 0;
+
+        while (productDownloads < productLimit) {
+            final QueryImpl query = new QueryImpl(filter);
+            query.setPageSize(productLimit);
+            query.setStartIndex(pageCount * productLimit + 1);
+            ++pageCount;
+
+            final QueryRequestImpl queryRequest;
+            if (sources != null && !sources.isEmpty()) {
+                queryRequest = new QueryRequestImpl(query, sources);
+            } else {
+                queryRequest = new QueryRequestImpl(query, true);
+            }
+            queryRequest.setProperties(new HashMap<>(CACHE_UPDATE_PROPERTIES));
+
+            final QueryResponse queryResponse = framework.query(queryRequest);
+
+            if (queryResponse.getResults()
+                    .isEmpty()) {
+                break;
+            }
+
+            final List<Entry<? extends ResourceRequest, String>> resourceRequests =
+                    queryResponse.getResults()
+                            .stream()
+                            .map(Result::getMetacard)
+                            .filter(isNonCachedResource)
+                            .map(metacard -> new SimpleEntry<>(new ResourceRequestById(metacard.getId()),
+                                    metacard.getSourceId()))
+                            .collect(toList());
+
+            for (Entry<? extends ResourceRequest, String> requestAndSourceId : resourceRequests) {
+                final ResourceRequest request = requestAndSourceId.getKey();
+                try {
+                    framework.getResource(request, requestAndSourceId.getValue());
+                    ++productDownloads;
+                } catch (IOException | ResourceNotFoundException | ResourceNotSupportedException e) {
+                    ++downloadErrors;
+                    LOGGER.warn("Could not download product for metacard [id={}]",
+                            request.getAttributeValue(),
+                            e);
+                }
+
+                printProgressAndFlush(start, productLimit, productDownloads);
+
+                if (productDownloads == productLimit) {
+                    break;
+                }
+            }
+        }
+
+        printProgressAndFlush(start, productDownloads, productDownloads);
+
+        console.println();
+        if (downloadErrors > 0) {
+            printColor(Ansi.Color.YELLOW,
```

Comment:
Is there a reason you don't want to use `printErrorMessage()` here? That will print this in RED, but this seems like it is an error condition to me.